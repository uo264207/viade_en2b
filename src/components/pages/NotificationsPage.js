import React from "react";
import "assets/css/NotificationsPage.css";
import CustomLoader from 'components/generic_components/CustomLoader';
import BurgerMenu from '../generic_components/BurgerMenu';


class NotificationsPage extends React.Component {

  constructor(props){
    super(props);
  
  this.state = {
    loading: false, //now, in the future will be false
    routes: "",
    comments:  [{
      title: "Perhaps...",
      text: "This is a comment, maybe, may, well..."
  }]
  };
}
  comments = [{
      title: "Perhaps...",
      text: "This is a comment, maybe, may, well..."
  },{
      title:"Well",
      text: "This is another comment"
  }];
  
 // componentDidMount() {
    //cache.default.getRoutes().then(rutas => {
   //   this.setState({ loading: false, comments: rutas });
   // });
        
 // }

  viewLoaded = comments =>{
   /* function search(){
        var value = document.getElementById("myInput").value;
        routes = routes.filter(item=>
          item.name.search(value)<0
        );
      }
    */

    return(
      
      <div className="bodyRoutes" id="outer-container">
      <main>
          <BurgerMenu 
          pageWrapId="page-wrap"
          container="outer-container"
          />
        <div className="App comments" id="page-wrap">
          <header className="bodyHeader"></header>
          <section className="sectionComments">
            <ul className="listComment">
              {comments.map((item, index)=>{
                return (
                  <li id={"comment"+index} key={index} className="liComment">
                    <div className="sectionComment">
                        <div className="commentInfo">
                            <p className="header"><i className="fa fas fa-bell"></i>{ item.title}</p>
                            <p className="description">{item.text}</p>
                        </div>
                        <div className="delete">
                            <span><i className="fa fas fa-trash"></i></span>
                        </div>
                    </div>
                  </li>
                );
              })} 
            </ul>
          </section>
        </div>
      </main>
    </div>
    );
    }
  viewCharge = ()=>{
    return(
      <div className="bodyComments" id="outer-container">
        <CustomLoader/>
       </div>
    );
  }
  render(){
    //const {loading} = this.state;

    return ( 
      <React.Fragment>
        {//loading ? <CustomLoader/> : this.viewLoaded(this.state.comments)
        }
        {this.viewLoaded(this.comments)}
      </React.Fragment>
  );
}
}

export default NotificationsPage;
