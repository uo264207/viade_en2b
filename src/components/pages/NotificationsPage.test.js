import React, {Component} from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow, mount } from 'enzyme';
import NotificationsPage from './NotificationsPage';

Enzyme.configure({ adapter: new Adapter() });

/////////////////////////

test('NotificationsPage',()=>{
    expect(NotificationsPage).toBeDefined();
});

/////////////////////////

test('Render Correct',()=>{
    const main = mount(<NotificationsPage/>);
    expect(main).toMatchSnapshot();
});

/////////////////////////

test('Render HamburgerMenu is Correct',()=>{
    const main = mount(<NotificationsPage/>);

    expect(main.find('BurgerMenu').exists()).toBe(true);
});

/////////////////////////

test('Render List is Correct',()=>{
    const main = mount(<NotificationsPage/>);

    expect(main.find('FloatingButton').exists()).toBe(false);
    expect(main.find('.listComment').exists()).toBe(true);
});


test('BurgerMenu change view',()=>{
    const main = mount(<NotificationsPage/>);

    const sideMenu = main.find('BurgerMenu');

    expect(sideMenu.find('#list-routes').at(1).props().href).toBe("#/routes");
});