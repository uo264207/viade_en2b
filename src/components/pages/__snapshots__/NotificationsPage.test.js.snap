// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Correct 1`] = `ReactWrapper {}`;
